src/types.jl

```diff
@@ -252,3 +269,59 @@ function Base.show(io::IO, bp::BreakpointRef)
     end
     print(io, ')')
 end
+
+# Possible types for breakpoint condition
+const Condition = Union{Nothing,Expr,Tuple{Module,Expr}}
+
+abstract type AbstractBreakpoint end
+
+same_location(::AbstractBreakpoint, ::AbstractBreakpoint) = false
+
+struct BreakpointSignature <: AbstractBreakpoint
+    f # Method or function
+    sig::Union{Nothing, Type}
+    line::Int # 0 is a sentinel for first statement
+    condition::Condition
+    enabled::Ref{Bool}
+    applications::Vector{BreakpointRef}
```

Comment:
Would `instances` be a better name for this field? Whatever you choose, obviously use the same name for `BreakpointFileLocation`.

Reply:
Sounds good.